fiddler/datadog_checks/fiddler/check.py

```diff
@@ -0,0 +1,164 @@
+
+        self.base_url = self.instance.get('url')
+        self.api_key = self.instance.get('fiddler_api_key')
+        self.org = self.instance.get('organization')
+
+        #            self.log.info("Connecting to : ", self.base_url)
+        #            self.log.info("with org id : ", self.org)
+        #            self.log.info("and auth key : ", self.api_key)
+
+        print("Connecting to : ", self.base_url)
+        print("with org id : ", self.org)
+        print("and auth key : ", self.api_key)
+
+        self.client = fdl.FiddlerApi(url=self.base_url, org_id=self.org, auth_token=self.api_key)
+
+    # Use self.instance to read the check configuration
+
+    def check(self, _):
+        # Iterate through the projects and the models and push data into Fiddler
+        project_path = ['list_projects', self.ORG_ID]
+        result_all = self.client.v1._call(project_path)
+        #        self.log.info("Projects: ", result_all["projects"])
+
+        start_time = (time.time() * 1000) - (bin_size * 1000)
+        end_time = time.time() * 1000
+        self.log.info("Start time is : ", start_time)
+        self.log.info("End time is : ", end_time)
+
+        # Iterate through all of the projects within the Fiddler instance and get the metrics
+        for project in result_all["projects"]:
+            models = project["models"]
+
+            # Iterate through all of the models within a project
+            for model in models:
+                self.log.info("Model: ", model["id"])
+                for metric in metrics_list:
+                    self.log.info("Metric is :", metric)
+                    json_request = {
+                        "metric": metric,
+                        "time_range_start": start_time,
+                        "time_range_end": end_time,
+                        "bin_size": bin_size,
+                        "prediction": '_',
+                    }
+                    agg_metrics_path = ['aggregated_metrics', self.ORG_ID, project["name"], model["id"]]
+                    self.log.info("ProjectModel: ", project["name"], model["id"], metric)
+                    # result = self.client._call(agg_metrics_path, json_request)
+
+                    hit_exception = False
+                    try:
+                        result = self.client.v1._call(agg_metrics_path, json_request)
+                    except Exception:
+                        self.log.info("Aggregated metrics exception : ", agg_metrics_path)
+                        self.log.info(
+                            "Project with no monitoring data. ProjectModel: ", project["name"], model["id"], metric
+                        )
+                        hit_exception = True
+
+                    if hit_exception:
+                        self.log.info("Hit the exception.")
+                        hit_exception = False
+                        continue
```

Comment:
I'm confused about why we need the `hit_exception` variable here? Why not:

```suggestion
                    try:
                        result = self.client.v1._call(agg_metrics_path, json_request)
                    except Exception:
                        self.log.info("Aggregated metrics exception : ", agg_metrics_path)
                        self.log.info(
                            "Project with no monitoring data. ProjectModel: ", project["name"], model["id"], metric
                        )
                        continue
```

Reply:
@alopezz - there is a chance that one model might throw an exception due to some reason. That should not prevent the operation to fail on the other models and hence we want to iterate through the rest of the models even if there is an exception on one of them.